package/kedro_viz/server.py

```diff
@@ -398,70 +414,114 @@ def format_pipeline_data(
                 "full_name": getattr(node, "_func_name", str(node)),
                 "tags": sorted(node.tags),
                 "pipelines": [pipeline_key],
+                "modular_pipelines": _expand_namespaces(node.namespace),
             }
             nodes_list.append(nodes[task_id])
         else:
             nodes[task_id]["pipelines"].append(pipeline_key)
 
         for data_set in node.inputs:
-            namespace = data_set.split("@")[0]
-            namespace_to_layer[namespace] = dataset_to_layer.get(data_set)
-            namespace_id = _hash(namespace)
+            dataset_namespace = data_set.split("@")[0]
+            if node.namespace:
+                namespace_to_modular_pipelines[dataset_namespace] = _expand_namespaces(
```

Comment:
Maybe extract `_expand_namespaces(node.namespace)` into a variable so you can re-use it in 3 places?